### PR TITLE
Basic Tutorial System

### PR DESCRIPTION
**Pull request by kevinsun-dev**
## Overview

Added a basic tutorial system that teaches:
 - Dragging stardust to the core
 - Avoiding stardust of a different color
 - Locking in
 - Opponent planets
 - Feedback system
 - Power ups
 - Win condition

## Changes Made

 - Removed TutorialMenu
 - Added TutorialScene

## Next Steps
 - Add Visual UI elements for understanding power-ups
 - More closely align the basic tutorial with the designs

## Screenshots

Screen Shot 2021-05-03 at 7 04 31 PM
Screen Shot 2021-05-03 at 7 05 34 PM

### CODE REVIEW

File: source/CIGameScene.cpp [hunk 1]
```diff
@@ -271,8 +271,14 @@ void GameScene::update(float timestep) {
...
     
     processSpecialStardust(dimen, _stardustContainer);
+    
+    /** Handle pause menu requests*/
     togglePause(_networkMessageManager->getGameState() == GameState::GamePaused);
     _pauseMenu->update();
+    if (_pauseMenu->_exitGame){
+        _winScene->setDisplay(false);
+        setActive(false);
+    }
```
Comment (henryz2000) [inline source/CIGameScene.cpp:281]: should set this to be a getter

File: source/CIPauseMenu.h [hunk 2]
```diff
@@ -52,6 +52,8 @@ class PauseMenu {
     std::shared_ptr<cugl::scene2::Button> _exitBtn;
 
 public:
+    /** Boolean setting request to exit game */
+    bool _exitGame;
```
Comment (henryz2000) [inline source/CIPauseMenu.h:56]: should not be public

**Review by henryz2000 — DISMISSED** [review]

File: source/CIGameScene.cpp (hunk 1 above)
Comment (kevinsun-dev) [reply source/CIGameScene.cpp]: Good feedback, fixed 👍

File: source/CIPauseMenu.h (hunk 2 above)
Comment (kevinsun-dev) [reply source/CIPauseMenu.h:55]: Yup, moved to protected. 👍

File: source/CIGameScene.cpp [hunk 3]
```diff
@@ -271,8 +271,14 @@ void GameScene::update(float timestep) {
     }
     
     processSpecialStardust(dimen, _stardustContainer);
+    
+    /** Handle pause menu requests*/
     togglePause(_networkMessageManager->getGameState() == GameState::GamePaused);
     _pauseMenu->update();
+    if (_pauseMenu->getExitGame()){
+        _winScene->setDisplay(false);
```
Comment (wlong123) [inline source/CIGameScene.cpp:279]: I think you want something along the lines of _pauseMenu->setDisplay(false)

**Review by wlong123 — APPROVED** [review]
LGTM! Just the one minor comment for exiting the game from the pause menu

**Pull request merged**